fabrics.c

```diff
@@ -1213,11 +1221,12 @@ int nvmf_config(const char *desc, int argc, char **argv)
 	char *subsysnqn = NULL;
 	char *transport = NULL, *traddr = NULL;
 	char *trsvcid = NULL, *hostnqn = NULL, *hostid = NULL;
-	char *hnqn = NULL, *hid = NULL;
+	_cleanup_free_ char *hnqn = NULL;
+	_cleanup_free_ char *hid = NULL;
```

Comment:
Pre-existing, but these variables look redundant with `hostnqn` and `hostid`, since they are assigned at the same time. Why do we need both?

Reply:
I think that's because ```nvmf_hostnqn_from_file()``` & ```nvmf_hostid_from_file()``` allocates memory while fetching the respective hostnqn & hostid strings. And while exiting this current function, one would ideally want to deallocate this memory but retain these string values at the same time, and that is accomplished by using these redundant variables ```hnqn``` & ```hid``` respectively.

Reply:
I'm not following. `hid` and `hnqn` aren't used anywhere and their values match `hostid` and `hostnqn`. So why not remove `hid` and `hnqn` and instead mark `hostid` and `hostnqn` as `_cleanup_free_`?

Reply:
`hostid` and `hostnqn` are origin from `argc`, while `nvme_hostid_from_file` and `nvme_hostnqn_from_file` allocated from the heap. We need to free the heap allocation, thus we have to track it via additional variables. Not nice, but unless we start to allocate all arguments on the heap we have to distinguish between the type of memory we operate on.

Reply:
> `hostid` and `hostnqn` are origin from `argc`

Ah, I missed the `NVMF_ARGS()` macro magic. Thanks.